src/main/kotlin/com/atlassian/performance/tools/virtualusers/ExploratoryVirtualUser.kt

```diff
@@ -40,10 +35,11 @@ internal class ExploratoryVirtualUser(
     }
 
     /**
-     * Repeats [actions] until the thread is interrupted.
+     * Repeats [actions] until [done] is `true`.
      */
-    fun applyLoad() {
-        shutdown.set(false)
+    fun applyLoad(
+        done: AtomicBoolean
+    ) {
         logger.info("Applying load...")
         logIn()
```

Comment:
You mean "unhang"?
It will un-hang it. We no longer block on `ExploratoryVirtualUser.applyLoad`, we `close` the output buffers regardless of what the VU threads are doing (e.g. even if they're hung on login).

The `done` is there only to keep VU threads of looping through the scenario after the `LoadSegment` is closed. Otherwise they'd only keep emitting errors. We might be fine with that, because none of the failed action metrics would be written to disk anyway (because the writers are closed), but I wanted to avoid unnecessary exception throwing in an infinite loop (it can make debugging more noisy and it's wasted work for the CPU).

Reply:
`done` is exactly like `shutdown`, but it's no longer global. It allows running independent `LoadTest`s in parallel within the same JVM (e.g. in integration tests).